streaming/src/buffer_pool/buffer_pool.h

```diff
@@ -0,0 +1,270 @@
+#pragma once
```

Comment:
consider the other c++ impl later, set the directory 'src/transfer' may be better

Reply:
> consider the other c++ impl later, set the directory 'src/transfer' may be better

It's no matter with transfer. Actually it's indenpendent.
BTW, this pr just keep same directory with internal version.